keps/sig-windows/20181221-windows-group-managed-service-accounts-for-container-identity.md

```diff
@@ -338,9 +338,9 @@ In the Beta phase, a new field `CredentialSpec String` will be added to `Windows
 
 #### Changes in Dockershim
 
-During Alpha, `dockerService.CreateContainer` function will be enhanced (under a feature flag: `WindowsGMSA`) to create a temporary file with a unique name on the host file system under path `C:\ProgramData\docker\CredentialSpecs\`. This file will be populated with the contents of `container.alpha.windows.kubernetes.io/gmsa-credential-spec` annotation in `CreateContainerRequest.ContainerConfig`. Beta onwards, `dockerService.CreateContainer` (under a feature flag: `WindowsGMSA`) will use the contents of `WindowsContainerSecurityContext.CredentialSpec` to populate the file.
+The GMSA credential spec will be passed to Docker through temporary entries in the Windows registry under SOFTWARE\Microsoft\Windows NT\CurrentVersion\Virtualization\Containers\CredentialSpecs. The registry entries will be created with unique key names that have a common prefix. The contents of the registry entries will be used to populate `HostConfig.SecurityOpt` with a credential spec file specification. The registry entries will be deleted as soon as `CreateContainer` has been invoked on the Docker client. An alternative implementation considered was to utilize files instead of registry entries but the path and drive where the files can be stored is hard to determine as there could be multiple installations of different versions of Docker engine under different directory paths. 
```

Comment:
do we need to secure the registry entries with any ACLs?

Reply:
They don't contain any secrets, so I don't think that's needed?

Reply:
no problem. we are good then